lib/SIL/IR/SILFunctionType.cpp

```diff
@@ -407,6 +407,7 @@ static CanSILFunctionType getAutoDiffDifferentialType(
     // result's convention is indirect.
     if (tl.isAddressOnly() && !isIndirectFormalResult(origResConv)) {
       switch (origResConv) {
+      case ResultConvention::Unowned:
```

Comment:
Note: we could make this logic handle all `ResultConvention` cases.

But with this change, the three common `ResultConvention` cases are all handed (`Unowned`, `Owned`, `Indirect`). I haven't seen the other cases (`UnownedInnerPointer`, `Autoreleased`) appear in practice.